lib/dependencies/HarmonyInitDependency.js

```diff
@@ -4,39 +4,74 @@
 */
 "use strict";
 
+/** @typedef {import('../Module')} Module **/
+/** @typedef {import('../Dependency')} Dependency **/
+/** @typedef {import("webpack-sources").ReplaceSource} ReplaceSource */
+/** @typedef {import("../RuntimeTemplate")} RuntimeTemplate */
+
+/** @typedef {import('../Dependency').DependencyTemplate} DependencyTemplate **/
+/** @typedef {import('./NullDependency').HarmonyInitDependencyTemplate} HarmonyDependencyTemplate **/
+
 const NullDependency = require("./NullDependency");
 
 class HarmonyInitDependency extends NullDependency {
+	/**
+	 * @param {Module} originModule - original module
+	 */
 	constructor(originModule) {
 		super();
+		/** @type {Module} */
 		this.originModule = originModule;
 	}
 
+	/**
+	 * @override
+	 */
 	get type() {
 		return "harmony init";
 	}
 }
 
 module.exports = HarmonyInitDependency;
 
+/**
+ * @implements {DependencyTemplate}
+ */
 HarmonyInitDependency.Template = class HarmonyInitDependencyTemplate {
+	/**
+	 * @param {HarmonyInitDependency} dep - dependency
+	 * @param {ReplaceSource} source - source
+	 * @param {RuntimeTemplate} runtime - runtime template
+	 * @param {Map<Function, DependencyTemplate | HarmonyDependencyTemplate>} dependencyTemplates - dependencies map
+	 * @override
+	 * @returns {void}
+	 */
 	apply(dep, source, runtime, dependencyTemplates) {
 		const module = dep.originModule;
+		/**
+		 * @typedef {number} IndexInList
+		 * @type {Array<{order: number, listOrder: IndexInList, dependency: Dependency, template: HarmonyDependencyTemplate}>}
+		 */
 		const list = [];
 		for (const dependency of module.dependencies) {
 			const template = dependencyTemplates.get(dependency.constructor);
+
 			if (
 				template &&
-				typeof template.harmonyInit === "function" &&
-				typeof template.getHarmonyInitOrder === "function"
+				typeof /** @type {HarmonyDependencyTemplate} */ (template).harmonyInit ===
```

Comment:
Only cast it once. Maybe in or after line 57.

Reply:
done